### ISSUE TITLE
Mac Voice Control Blocked

### ISSUE THREAD

**Issue by davidkastner** [microsoft/vscode#106124]
- VSCode Version: 1.48.2
- OS Version: Catalina 10.15.6

Steps to Reproduce:

1. Mac > System preferences > Accessibility > Voice Control > Enable Voice Control
2. Dictate in VS Code

Does this issue occur when all extensions are disabled?: Yes

Apple’s new “Voice Control” software released with Catalina is barely functioning and glitchy on VS Code. I can get it to work sometimes by pre-clicking on buttons or the menu but nothing consistent. I can open it with "open code" and close it with "quit code" but everything else is glitchy. I’ve tried it on two MacBook pros, including the most recent release and it does not work for either. I believe this may be a serious problem as many people with disabilities such as paralysis and ALS depend on the new “Voice Control” software. I found a previous mention of Dragon but this a separate issue since Voice Control is part of the Mac OS and therefore a loss of function bug and not a gain of function feature.

**Comment by chuckthepiranha** [comment]
This is a problem for me as well. It isn't that it is disabled, it's functionality is just inconsistant. It only works like 10% of the time.

**Comment by isidorn** [comment]
Related issue https://github.com/microsoft/vscode/issues/40976

Thanks for filling this issue. Some questions:
1. Does "Voice Control" software work better with Google Chrome?
2. If the answer to previous question is "yes". If you use VS Code insiders is there any improvement https://code.visualstudio.com/insiders/
3. Can you be more specific about what exactly does not work. Just a simple example would do
4. How is this different than this comment https://github.com/microsoft/vscode/issues/40976#issuecomment-663808075

**Comment by davidkastner** [comment]
1. Voice Control works better in Chrome. For example, if I say "hashtag this is a python script" in the main navigation bar, my mac will dictate it "# this is a python script" perfectly. However, say "hashtag this is a python script" in VS code and nothing happens.
2. I just downloaded and installed the insider's release and don't see a difference.
3. To be more specific, "voice typing" is not working in VS Code. However, commands do work, which makes it seem like a bug since Voice Control works sometimes. For example, open a new document in VS code and say "hashtag this is a python script" nothing happens. Next type "# this is a python script" in VS code and say "select all," "copy that", "move down," "press return key" and "paste that." VS Code will copy the line and paste it without any problems.
4. I read over this comment before posting mine but it is referring to a specific command called "show numbers," which doesn't work in hardly any apps so I would not expect it to work in VS Code. Voice typing, however, works in lots of apps from Word to fringe apps like PyMol and I found it surprising that it didn't work in VS Code.

**Comment by isidorn** [comment]
@davidkastner thanks for providing more details.
I have just enabled Voice Control and here are my findings (can you please verify since I used Voice Control for the first time):

1. I see the same problem as you do when using VS Code
2. I see the same problem when using Chrome with a `textArea` element. For example try this site `https://bocupp-microsoft.github.io/sample-code/textarea/setselectionrange-performance.html` Typing in the text box does not work - same as VS Code

So the underlying issue is "Voice Control does not work with textArea in Chrome" and we should file this against Chrome. Since VS Code uses a hidden textArea and we use Chrome as an underlying engine.
Can you file this issue to chomre or would you prefer that I do it?

**Comment by isidorn** [comment]
I have filled this issue in Chromium https://bugs.chromium.org/p/chromium/issues/detail?id=1147866

**Comment by isidorn** [comment]
This is fixed in Chrome 89. Thus will be fixed in electron 12.
@deepak1556 if you have time is it possible to backport https://chromium.googlesource.com/chromium/src/+/95ffeb855c7b84cc6654ff4c4c9e7303785dc241

Otherwise we can wait for the Electron 12 adoption

**Comment by deepak1556** [comment]
Seems minimal, can backport it.

**Comment by deepak1556** [comment]
Updates, the CL linked in https://github.com/microsoft/vscode/issues/106124#issuecomment-789022966 has ben reverted https://bugs.chromium.org/p/chromium/issues/detail?id=1143047#c10 due to entering accessibility mode when not intended. It is now superseded by https://bugs.chromium.org/p/chromium/issues/detail?id=1143047#c14

But the bug is blocked on https://bugs.chromium.org/p/chromium/issues/detail?id=1132255 which is waiting on https://chromium-review.googlesource.com/c/chromium/src/+/2716865. I will backport the fix once the bug is closed.

**Comment by isidorn** [comment]
@deepak1556 thanks for details. Let's assign to on-deck then until the fix is available in Chromium.

**Comment by isidorn** [comment]
In a related issue a user reported that this should now be fixed with Mac OS X (Big Sur 11.2.3)

**Comment by motclem** [comment]
I am using big Sur 11.3.1And I'm experiencing difficulties using voice controlSimply To type Into The editor.

**Comment by isidorn** [comment]
@motclem thanks for letting us know, so do you experience the same thing explained in the original comment? If not could you share some repro steps?

**Comment by motclem** [comment]
Yes I believe it is the same as the original comment simplify wake voice control up And then tried to dictate into VSC nothing happens.

**Comment by falenas108** [comment]
Going with what was said above, Voice control works OK with chrome both for the show numbers command and text typing, but does not work on visual studio code.

**Comment by motclem** [comment]
As that is correct I am having the same problem as in the opening post weird Voice control tent on and a awake if I try to dictate anything into a new file in visual studio code there is no output written to the screen.

**Comment by VSCodeTriageBot** [comment]
This issue has been closed automatically because it needs more information and has not had recent activity. See also our [issue reporting](https://aka.ms/vscodeissuereporting) guidelines.

Happy Coding!
